calc_elasticscattering.py

```diff
@@ -6,43 +6,38 @@
 
 eV = 1.60217657e-19
 
-# output folder
-folder = 'data/ElasticScattering'
-if not os.path.exists(folder):
-    os.makedirs(folder)
+def process(field):
+    """ calculate the interaction rates for a given photon field"""
 
-gamma = np.logspace(6, 14, 201)  # tabulated UHECR Lorentz-factors
+    # output folder
+    folder = 'data/ElasticScattering'
+    if not os.path.exists(folder):
+        os.makedirs(folder)
 
-fields = [
-    photonField.CMB(),
-    photonField.EBL_Gilmore12(),
-    photonField.URB_Protheroe96()
-    ]
+    gamma = np.logspace(6, 14, 201)  # tabulated UHECR Lorentz-factors
 
+    # load cross section data from TALYS
+    ddir = 'tables/PD_Talys1.8_Khan/'
+    eps = np.genfromtxt(ddir + 'eps_elastic.txt') * eV * 1e6  # nuclear rest frame photon energies [J]
+    data = np.genfromtxt(ddir + 'xs_elastic.txt', dtype=[('Z', int), ('N', int), ('xs', '%if8' % len(eps))])
 
-# load cross section data from TALYS
-ddir = 'tables/PD_Talys1.8_Khan/'
-eps = np.genfromtxt(ddir + 'eps_elastic.txt') * eV * 1e6  # nuclear rest frame photon energies [J]
-data = np.genfromtxt(ddir + 'xs_elastic.txt', dtype=[('Z', int), ('N', int), ('xs', '%if8' % len(eps))])
+    # only consider TALYS cross sections for A >= 12
+    idx = (data['Z'] + data['N']) >= 12
+    data = data[idx]
 
-# only consider TALYS cross sections for A >= 12
-idx = (data['Z'] + data['N']) >= 12
-data = data[idx]
+    # factor out the principal scaling given by the TRK formula: sigma_int ~ Z*N/A
+    data['xs'] /= (data['Z'] * data['N'] / (data['Z'] + data['N']))[:, np.newaxis]
 
-# factor out the principal scaling given by the TRK formula: sigma_int ~ Z*N/A
-data['xs'] /= (data['Z'] * data['N'] / (data['Z'] + data['N']))[:, np.newaxis]
+    # pad cross sections to next larger 2^n + 1 tabulation points for Romberg integration
+    eps = iR.romb_pad_logspaced(eps, 513)
+    xs = np.array([iR.romb_pad_zero(x, 513) for x in data['xs']]) * 1e-31
 
-# pad cross sections to next larger 2^n + 1 tabulation points for Romberg integration
-eps = iR.romb_pad_logspaced(eps, 513)
-xs = np.array([iR.romb_pad_zero(x, 513) for x in data['xs']]) * 1e-31
 
 
-for field in fields:
-    print(field.name)
 
     # calculate the interaction rate, averaged over all isotopes
     rate = np.mean([iR.calc_rate_eps(eps, x, gamma, field) for x in xs], axis=0)
-    fname = folder + '/rate_%s.txt' % field.name.split('_')[0]
+    fname = folder + '/rate_%s.txt' % field.name[:3]
```

Comment:
Why did you change the from splitting at "_" to picking the first 3 characters?

Reply:
The current implementation in CRPropa takes only 3 characters. I changed this to be consistent in the case that users does not stick to the naming with an abbreviation of 3 characters.